django-s3-file-field-client/src/client.ts

```diff
@@ -68,7 +82,7 @@ export default class S3FFClient {
    * @param parts the list of parts describing how to break up the file
    * @returns UploadedPart[]
    */
-  protected async uploadParts(file: File, parts: PartInfo[]): Promise<UploadedPart[]> {
+  protected async uploadParts(file: File, parts: PartInfo[], options: UploadOptions = {}): Promise<UploadedPart[]> {
```

Comment:
There is more work to the progress than just uploading the parts. Each of the requests to the server requests also represents some amount of asynchronous progress.

Reply:
True, though in the past I've been happy to just report indeterminate progress while those things are happening, rather than trying to fake quantified progress. I notice in the other library you guys use a statically fixed percentage as an approximation, but I'm not wild about that -- in my case, I'll be uploading files that are hundreds of MB, maybe over 1GB each, and in other cases it'll be tiny stuff of a couple KB. If you really want to go the route of determinate progress throughout, let's at least base the amount dynamically on the file size.